packages/currency/src/token.ts

```diff
@@ -22,11 +22,14 @@ export class Token extends Currency {
    */
   static from(symbolOrAddress: string): Token {
     try {
-      const currencyFromSymbol = this.fromSymbol(symbolOrAddress);
+      const currencyFromSymbol = this.fromSymbol(
+        symbolOrAddress.split('-')[0],
+        symbolOrAddress.split('-')[1],
+      );
       return new Token(
         currencyFromSymbol.value,
         currencyFromSymbol.type,
-        symbolOrAddress,
+        symbolOrAddress.split('-')[0],
```

Comment:
I don't  get this part

if I do Token.from("FAU-rinkeby") it will work, but with Token.from("0xabcd") it won't?

Reply:
There is a try catch, we first try to guess from a token code. If it fails, we try an address.